### PR TITLE
feat: add C implementation for `math/base/special/coth`

### LINKED ISSUE #754: [RFC]: Add C implementation for `@stdlib/math/base/special/coth`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/coth`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/coth).

```c
double stdlib_base_coth( const double x );
```

### Related Issues

#752 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by lovelindhoni**
Resolves #754  .

## Description

This pull request adds native C implementation for `@stdlib/math/base/special/coth`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #754
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/coth/LICENSE [hunk 1]
```diff
@@ -0,0 +1,196 @@
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/LICENSE]: Please revert this

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c [hunk 2]
```diff
@@ -0,0 +1,135 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 10.0*rand_double() ) - 5.0;
+		y = coth( x );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c]: `coth` is not present in `math.h` you may remove this file.

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/cephes/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	return (double)r / ( (double)RAND_MAX + 1.0 );
+}
+
+/**
+* Runs a benchmark.
+*
+* @return elapsed time in seconds
+*/
+double benchmark() {
+	double elapsed;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/cephes/benchmark.c]: You may wish to revert this with same reason.

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/native/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,136 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/native/benchmark.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/coth/include/stdlib/math/base/special/coth.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/include/stdlib/math/base/special/coth.h]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/coth/src/main.c [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+* double out = stdlib_base_coth( 0.0 );
+* // returns INFINITY
+*/
+
+double stdlib_base_coth( const double x ) {
+    if (x == 0.0) {
+        return INFINITY;
+    }
+    return 1.0 / stdlib_base_tanh(x);
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/src/main.c]:
```suggestion
double stdlib_base_coth( const double x ) {
    return 1.0 / stdlib_base_tanh(x);
}
```
I'll keep C implementation consistent with javascript.

File: lib/node_modules/@stdlib/math/base/special/coth/src/main.c [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/src/main.c]:
```suggestion
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Thanks @lovelindhoni! Good direction, apply code review and this will be ready to merge.

**Comment by lovelindhoni** [comment]
@Pranavchiku,  the requested changes have been made

**Pull request closed without merging**